File: TouchKitOC/TouchKitOC/RelatedFiles/TOUCHOverlayWindow.m
//
//  TOUCHOverlayWindow.m
//  TouchKitOC
//
//  Created by travelsky on 2018/4/12.
//  Copyright © 2018年 wangdicen. All rights reserved.
//

#import "TOUCHOverlayWindow.h"
#import "TOUCHkitView.h"

@implementation TOUCHOverlayWindow

- (void)sendEvent:(UIEvent *)event
{
    //截取UIWindow的事件分类存入集合.
    NSSet *touches = [event allTouches];
    NSMutableSet *began =  nil;
    NSMutableSet *moved = nil;
    NSMutableSet *ended = nil;
    NSMutableSet *cancelled = nil;
    
    for (UITouch *touch in touches) {
        switch ([touch phase]) {
            case UITouchPhaseBegan:
                if (!began) {
                    began = [NSMutableSet set];
                }
                [began addObject:touch];
                break;
            case UITouchPhaseMoved:
                if (!moved) {
                    moved = [NSMutableSet set];
                }
                [moved addObject:touch];
                break;
            case UITouchPhaseEnded:
                if (!ended)
                {
                    ended = [NSMutableSet set];
                }
                [ended addObject:touch];
                break;
            case UITouchPhaseCancelled:
                if (!cancelled)
                {
                    cancelled = [NSMutableSet set];
                }
                [cancelled addObject:touch];
                
            default:
                break;
        }
    }
    
    //更具事件类型去调用touchkitview单例的touch事件.
    if (began)
        [[TOUCHkitView sharedInstance] touchesBegan:began withEvent:event];
    if (moved)
        [[TOUCHkitView sharedInstance] touchesMoved:moved withEvent:event];
    if (ended)
        [[TOUCHkitView sharedInstance] touchesEnded:ended withEvent:event];
    if (cancelled)
        [[TOUCHkitView sharedInstance] touchesCancelled:cancelled withEvent:event];
    
    [super sendEvent: event];    
    
}

@end

File: TouchKitSwift/TouchKitSwift/RelatedFiles/TOUCHOverlayWindow.swift
//
//  File.swift
//  TouchKitSwift
//
//  Created by travelsky on 2018/4/12.
//  Copyright © 2018年 wangdicen. All rights reserved.
//

import Foundation
import UIKit

class TOUCHOverlayWindow: UIWindow {
    override func sendEvent(_ event: UIEvent) {
        let touches:Set = event.allTouches!;
        var began:Set = Set<UITouch>();
        var moved:Set = Set<UITouch>();
        var ended:Set = Set<UITouch>();
        var cancelled:Set = Set<UITouch>();
        
        for touch in touches {
            switch(touch.phase){
            case .began:
                began.insert(touch);
                break;
            case .moved:
                moved.insert(touch);
                break;
            case .ended:
                ended.insert(touch);
                break;
            case .cancelled:
                cancelled.insert(touch);
                break;
            default:
                break;
            }
        }
        
        
        if began.count > 0{
            TOUCHkitView.shareInstance.touchesBegan(began, with:event);
        }
        if moved.count > 0
        {
            TOUCHkitView.shareInstance.touchesMoved(moved, with: event);
        }
        if ended.count > 0 {
            TOUCHkitView.shareInstance.touchesEnded(ended, with: event);
        }
        
        super.sendEvent(event);
    }
}
